File: src/types/video.ts
export interface Video {
  id: number;
  folder_id: number;
  path: string;
  filename: string;
  hash: string;
  duration: number; // seconds
  width: number; // pixels
  height: number; // pixels
  size: number; // bytes
  codec: string | null;
  framerate: number | null; // fps
  thumbnail_path: string | null;
  thumbnail_count: number;
  rating: number; // 0-5
  is_favorite: number; // 0 or 1
  created_at: string; // ISO 8601
  updated_at: string; // ISO 8601
  scanned_at: string; // ISO 8601
}

export interface Folder {
  id: number;
  path: string;
  recursive: boolean;
  created_at: string; // ISO 8601
  updated_at: string; // ISO 8601
}

export interface ScanResult {
  videos_found: number;
  videos_added: number;
  videos_updated: number;
  errors: string[];
}

export interface ScanProgress {
  current: number;
  total: number;
  current_file: string;
}

export interface VideoMetadata {
  duration: number;
  width: number;
  height: number;
  codec: string;
  framerate: number;
}

File: src-tauri/src/models/video.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

/// Video model representing a video file with metadata
#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Video {
    pub id: i64,
    pub folder_id: i64,
    pub path: String,
    pub filename: String,
    pub hash: String,
    pub duration: i64,
    pub width: i64,
    pub height: i64,
    pub size: i64,
    pub codec: Option<String>,
    pub framerate: Option<f64>,
    pub thumbnail_path: Option<String>,
    pub thumbnail_count: i64,
    pub rating: i64,
    pub is_favorite: i64,
    pub created_at: String,
    pub updated_at: String,
    pub scanned_at: String,
}

/// Input struct for inserting a new video
#[derive(Debug, Clone)]
pub struct VideoInsert {
    pub path: String,
    pub filename: String,
    pub hash: String,
    pub duration: i64,
    pub width: i64,
    pub height: i64,
    pub size: i64,
    pub codec: Option<String>,
    pub framerate: Option<f64>,
}
